storage/remote/queue_manager.go

```diff
@@ -402,13 +410,16 @@ func (t *QueueManager) StoreSeries(series []record.RefSeries, index int) {
 // stored series records with the checkpoints index number, so we can now
 // delete any ref ID's lower than that # from the two maps.
 func (t *QueueManager) SeriesReset(index int) {
+	t.seriesMtx.Lock()
+	defer t.seriesMtx.Unlock()
 	// Check for series that are in segments older than the checkpoint
 	// that were not also present in the checkpoint.
 	for k, v := range t.seriesSegmentIndexes {
 		if v < index {
 			delete(t.seriesSegmentIndexes, k)
 			releaseLabels(t.seriesLabels[k])
 			delete(t.seriesLabels, k)
+			delete(t.droppedSeries, k)
```

Comment:
Was the absence of this line a bug (even before this commit)?

Reply:
Yes, I have a note about it in the commit message/pull request description. Should I break it out into a different commit or even PR?

Reply:
No need, I just didn't read the description. Thanks.